### PR TITLE
6 most analyzers dont work with third column or more

### PR DESCRIPTION
**Pull request by JAWalmsley**

### CODE REVIEW

**Comment by JAWalmsley** [comment]
As far as I can tell it works exactly the same as before, but now analyzers can be applied to datasets with more than 2 columns (GPS date stuff is the only example right now).

Also fixed a big in [d9124e7](https://github.com/McMaster-Baja-Racing/Better-Data-Viewer/pull/55/commits/d9124e7d89c7b95acf5ab576df3da40d39f25d55) where the output filename for accelcurve had a "/" in it so the backend would error trying to read it

**Review by gr812b — CHANGES_REQUESTED** [review]
Only major change is to the linear interpolator, 

also worth looking into in the future refactoring how the data is passed to /analyze on the controller, perhaps we can structure it in a similar manner to the /files method. Nothing on this needs to be changed as of now though

File: API/src/main/java/backend/API/analyzer/LinearInterpolaterAnalyzer.java [hunk 1]
```diff
@@ -32,10 +32,16 @@ public List<List<String>> linearInterpolate(List<List<String>> data1, List<List<
...
+        
+
         // Add header
-        dataPoint.add(data1.get(0).get(0));
-        dataPoint.add(data1.get(0).get(1));
-        dataPoint.add(data2.get(0).get(1));
+        dataPoint.add(data1.get(0).get(independentColumn));
+        dataPoint.add(data1.get(0).get(dependentColumn1));
+        dataPoint.add(data2.get(0).get(dependentColumn2));
         dataPoints.add(dataPoint);
```
[the remark this answers is not in the document]
Comment (JAWalmsley) [reply API/src/main/java/backend/API/analyzer/LinearInterpolaterAnalyzer.java:46]: Should be doing this now, enjoy that 60 char long filename

File: API/src/main/java/backend/API/analyzer/LinearMultiplyAnalyzer.java [hunk 2]
```diff
@@ -16,8 +16,8 @@ public class LinearMultiplyAnalyzer extends Analyzer {
 
     private static final DecimalFormat df = new DecimalFormat("0.00");
 
-    public LinearMultiplyAnalyzer(String[] inputFiles, String[] outputFiles, double m, double b) {
-        super(inputFiles, outputFiles);
+    public LinearMultiplyAnalyzer(String[] inputFiles, String[] inputColumns, String[] outputFiles, double m, double b) {
```
[the remark this answers is not in the document]
Comment (JAWalmsley) [reply API/src/main/java/backend/API/analyzer/LinearMultiplyAnalyzer.java:19]: just fixed

File: API/src/main/java/backend/API/analyzer/LinearMultiplyAnalyzer.java [hunk 3]
```diff
@@ -40,17 +40,20 @@ public List<List<String>> linearMultiply(List<List<String>> data, double m, doub
...
 
         //loop through the data and multiply the second coloum by m and then add b
         for (int i = 1; i < data.size(); i++) {
-            dataPoint.add(data.get(i).get(0));
-            dataPoint.add(df.format(Double.parseDouble(data.get(i).get(1)) * m + b));
+            dataPoint.add(data.get(i).get(independentColumn));
+            dataPoint.add(df.format(Double.parseDouble(data.get(i).get(dependentColumn)) * m + b));
             dataPoints.add(dataPoint);
             dataPoint = new ArrayList<String>(2);
         }
```
[the remark this answers is not in the document]
Comment (JAWalmsley) [reply API/src/main/java/backend/API/analyzer/LinearMultiplyAnalyzer.java:59]: fixed :)

File: API/src/main/java/backend/API/analyzer/LinearInterpolaterAnalyzer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (JAWalmsley) [reply API/src/main/java/backend/API/analyzer/LinearInterpolaterAnalyzer.java:46]: [0387616](https://github.com/McMaster-Baja-Racing/Better-Data-Viewer/pull/55/commits/0387616631f7b3fe85c104fea3565691a74cb668)

File: API/src/main/java/backend/API/analyzer/LinearMultiplyAnalyzer.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (JAWalmsley) [reply API/src/main/java/backend/API/analyzer/LinearMultiplyAnalyzer.java:19]: [b1ea9a4](https://github.com/McMaster-Baja-Racing/Better-Data-Viewer/pull/55/commits/b1ea9a429a9c7228461672b43300eab08a690ac6)

File: API/src/main/java/backend/API/analyzer/LinearMultiplyAnalyzer.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (JAWalmsley) [reply API/src/main/java/backend/API/analyzer/LinearMultiplyAnalyzer.java:59]: [bab1908](https://github.com/McMaster-Baja-Racing/Better-Data-Viewer/pull/55/commits/bab190840c40de3896a3a03752be756883dd5f17)

File: front-end/src/components/Chart.js [hunk 4]
```diff
@@ -37,8 +37,12 @@ const Chart = ({ chartInformation }) => {
     const [fileNames, setFileNames] = useState([]);
 
     const getFile = async (inputFiles, outputFiles, analyzerOptions, liveOptions, columnInfo) => {
+        // TODO: Remove debug code
```
Comment (gr812b) [inline front-end/src/components/Chart.js:40]: Worth removing soon

**Review by gr812b — CHANGES_REQUESTED** [review]
Overall looks great, I'm just curious on the behaviour if we can merge main into this branch and test with XYColour. Looks like it should make sense to fetch, but not 100% certain

**Comment by JAWalmsley** [comment]
Merge works, but workflow for linear interpolation should probably be reworked because right now we need to first make the file two columns at a time in Line, then graph it in XYColour

**Pull request merged**